Allow adding and removing attributes  in Context (#502)

uoe-pjackson opened the pull request:
Fixes #502 

A bit of a hack with an extra augeas was required to add attributes in <Context> element if there are no existing attributes.

This is my first puppet module pull request so apologies if I've got anything wrong.

Thanks!

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/puppetlabs/puppetlabs-tomcat?pullRequest=503) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/puppetlabs/puppetlabs-tomcat?pullRequest=503) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/puppetlabs/puppetlabs-tomcat?pullRequest=503) it.</sub>

uoe-pjackson commented:
Looks like the new tomcat version brought with them 1.3.2 commons-daemon-native instead of the previous 1.3.1

Didn't think my change should have broken the acceptance tests in such a big way!

uoe-pjackson commented:
I've rebased from main. 

I've fixed the tests by bumping up the commons-daemon-native version. I meant to put that into a separate PR but accidentally committed into this branch..

jordanbreen28 commented:
Re-kicking tests

uoe-pjackson commented:
@jordanbreen28 Rebased, all tests successful apart from mend test failing due to `Mandatory parameters apiKey/licenseKey were not set` Thanks!

jordanbreen28 reviewed (approved):
@uoe-pjackson mend scan is expected to fail, not to worry!
Merged :-) thanks again
